workflow-templates/psake-ci.yml

```diff
@@ -1,29 +1,18 @@
 # In the last step that publishes a GitHub Release, be sure to adjust
 # the 'files' argument with the correct path of the built artifacts
-# that should be included in the release. Typically: build/Latest
+# that should be included in the release. The file should be an already packaged zip file.
 #
-# Also, if you need to upload build artifacts then you can uncomment
-# the "Upload artifact" build step. Remember to replace the placeholders
-# <project-name>-<version> with the correct artifact name
 name: psake CI
 
 # What should trigger this workflow to be executed
 on:
   # Define which push events should trigger this workflow
   push:
-    # Trigger this workflow when pushing to either the master or develop branch
+    # Trigger this workflow when new tag is created
     # Adjust according to your project
-    branches:
```

Comment:
I think we should not remove `master` and `develop`, simply because that is how people have been using `psake ci` in the past.

Reply:
Alright, so I think there might be some confusion regarding the pipeline. I think there are two scenarios:

1) Run the `psake ci` to create a release version. 
2) Run the `psake ci` to deploy to some environment (most of them were creating octopus packages)

The first scenario produces an artifact, while the second one does not. Currently, the pipeline does (or could be doing) both of the things.

Now, I could add the if statement to every single step that leads to the artifact being produced:
`Add git tag to the environment`
`Zip release`
`Create release (if the tag was pushed)` (if statement already present)
but I feel that this is slightly odd as I see no difference between `psake tryCi` and `psake Ci`.

TryCi vs CI should either differ in the artifact being produced in the latter one, or in case they are two different processes, in their pipelines. Therefore, upon push to a `master` or `develop` branch, you should only be verifying if the solution builds, which can be achieved via `psake tryCi`.
If you want to deploy to some environment, you should maybe use `psake ci`, which runs on develop and master branch and deploys to some environment without producing the artifact.
Lastly, a pipeline called `psake release` (or something like that, but could also reuse `psake ci`) runs the psake command then gets the tag and produces an artifact Github Release entry.